packages/sync/src/conflicts/conflictLink.ts

```diff
@@ -15,30 +14,28 @@ export const conflictLink = (config: DataSyncConfig): ApolloLink => {
       for (const err of graphQLErrors) {
         if (err.extensions) {
           // TODO need to add flag to check if conflict was resolved on the server
-          if (err.extensions.exception.type === CONFLICT_ERROR) {
-            return err.extensions.exception.data;
+          if (err.extensions.exception.conflictInfo) {
+            return err.extensions.exception.conflictInfo;
           }
         }
       }
     }
   };
-  /**
-   * Fetch the conflict strategy if one is provided, if not return client wins.
-   */
-  const getConflictStrategy = (): ConflictResolutionStrategy => {
-    if (config.conflictStrategy) {
-      return config.conflictStrategy;
-    } else {
-      return strategies.diffMergeClientWins;
-    }
-  };
 
   return onError(({ operation, forward, graphQLErrors }) => {
     const data = getConflictData(graphQLErrors);
-    if (data) {
-      const resolvedConflict = getConflictStrategy()(data, operation.variables);
-      // TODO Notify
-      resolvedConflict.version = data.version;
+    if (data && config.conflictStrategy && config.conflictStateProvider) {
+      let resolvedConflict;
+      if (data.resolvedOnServer) {
+        resolvedConflict = data.serverData;
```

Comment:
I'm not too sure about this logic, so if the server resolved the conflict, we are going to get it here, call `nextState` and then call `forward` on the already resolved conflict right? Am I missing something here?

Reply:
Definitely bug. Fixed that.